L1Trigger/L1TMuonOverlapPhase2/interface/LutNetworkFixedPointCommon.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * LutNetworkCommon.h
+ *
+ *  Created on: Jan 13, 2023
+ *      Author: kbunkow
+ */
+
+#ifndef INTERFACE_LUTNETWORKFIXEDPOINTCOMMON_H_
+#define INTERFACE_LUTNETWORKFIXEDPOINTCOMMON_H_
+
+#include <ap_fixed.h>
+#include <ap_int.h>
+#include <array>
+#include <vector>
+#include <limits>
+
+#include <boost/property_tree/ptree.hpp>
+
+namespace lutNN {
+
+  template <int W, int I>
+  const ap_ufixed<W, I> max_ap_ufixed() {
+    static_assert(I < 64, "this max_ap_ufixed works only for I < 64");
+    return ap_ufixed<W, I, AP_RND, AP_SAT>(std::numeric_limits<uint64_t>::max());
+    //AP_SAT Saturate the value to the maximum value in case of overflow
+  }
+
+  template <int W, int I>
+  const ap_fixed<W, I> max_ap_fixed() {
+    static_assert(I < 64, "this max_ap_ufixed works only for I < 64");
+    return ap_fixed<W, I, AP_RND, AP_SAT>(std::numeric_limits<uint64_t>::max());
+    //AP_SAT Saturate the value to the maximum value in case of overflow
+  }
+
+#define PUT_VAR(tree, keyPath, var) tree.put((keyPath) + "." + #var, (var));
+
+#define CHECK_VAR(tree, keyPath, var)                 \
+  if ((var) != tree.get<int>((keyPath) + "." + #var)) \
+    throw std::runtime_error((keyPath) + "." + #var + " has different value in the file then given");
```

Comment:
I would, in general, avoid modification of any syntax via `define`: https://cms-sw.github.io/cms_coding_rules.html. I get that this is a macro and is unlikely to cause issues, but I would look for any other way to handle this first just to be cautious.

Reply:
Replacing macros by inline functions.

Reply:
Yes, exactly. It was also my conclusion that without preprocessor "stringify" #